### PR TITLE
Fix errcheck issue for saml.go file

### PR DESCRIPTION
**Pull request by TheInvincibleRalph**
#### Ticket Link

Fixes https://github.com/mattermost/mattermost/issues/28523

```release-note
NONE
```

### CODE REVIEW

File: server/channels/api4/saml.go [hunk 1]
```diff
@@ -46,7 +46,10 @@ func getSamlMetadata(c *Context, w http.ResponseWriter, r *http.Request) {
 
 	w.Header().Set("Content-Type", "application/xml")
 	w.Header().Set("Content-Disposition", "attachment; filename=\"metadata.xml\"")
-	w.Write([]byte(metadata))
+	if _, err := w.Write([]byte(metadata)); err != nil {
+		c.Err = model.NewAppError("getSamlMetadata", "api.admin.saml.get_metadata.app_error", nil, "", http.StatusInternalServerError).Wrap(err)
```
Comment (hanzei) [inline server/channels/api4/saml.go:50]: If `w.Write` fails, there is likely an issue with the HTTP connection. It's not useful to then send an error to the client. Let's just log the error.

File: server/channels/api4/saml.go [hunk 2]
```diff
@@ -135,7 +150,13 @@ func addSamlIdpCertificate(c *Context, w http.ResponseWriter, r *http.Request) {
 	}
 
 	auditRec := c.MakeAuditRecord("addSamlIdpCertificate", audit.Fail)
-	defer c.LogAuditRec(auditRec)
+	defer func() {
+		if c.Err == nil {
+			auditRec.Success()
```
Comment (hanzei) [inline server/channels/api4/saml.go:155]: The function calls `audit.Success` before returning. Why does it need to be called here as well?

**Review by hanzei — CHANGES_REQUESTED** [review]
Thanks for the contribution:+1:

File: server/channels/api4/saml.go (hunk 2 above)
Comment (TheInvincibleRalph) [reply server/channels/api4/saml.go:155]: I guess the defer function I added is not even needed since `c.LogAuditRec(auditRec)` is expected to be called when the function exits (regardless of success or failure). It would be better it stayed at the previous state it was.  Am I correct?

Comment (TheInvincibleRalph) [reply server/channels/api4/saml.go]: I have updated the PR based on your review.

File: server/channels/api4/saml.go [hunk 3]
```diff
@@ -46,7 +46,10 @@ func getSamlMetadata(c *Context, w http.ResponseWriter, r *http.Request) {
 
 	w.Header().Set("Content-Type", "application/xml")
 	w.Header().Set("Content-Disposition", "attachment; filename=\"metadata.xml\"")
-	w.Write([]byte(metadata))
+	if _, err := w.Write([]byte(metadata)); err != nil {
+		c.Logger.Warn("Error while writing response", mlog.Err(err))
+		return
```
Comment (hanzei) [inline server/channels/api4/saml.go:51]: Nit pick: Unnessary return

File: package-lock.json
Comment (hanzei) [inline package-lock.json:1]: Please remove the unrelated changes

**Review by hanzei — CHANGES_REQUESTED** [review]
Can you please remove the unrelated changes that add newlines in `server/channels/api4/saml.go`? Thanks!

**Comment by TheInvincibleRalph** [comment]
Changes made, @hanzei

File: package-lock.json
Comment (hanzei) [inline package-lock.json:1]: The unrelated changes are still part of this PR

File: server/channels/api4/saml.go [hunk 4]
```diff
@@ -83,6 +85,7 @@ func addSamlPublicCertificate(c *Context, w http.ResponseWriter, r *http.Request
 
 	auditRec := c.MakeAuditRecord("addSamlPublicCertificate", audit.Fail)
 	defer c.LogAuditRec(auditRec)
+
```
Comment (hanzei) [inline server/channels/api4/saml.go:88]: Please remove this unrelated changes.

**Review by hanzei — CHANGES_REQUESTED** [review]
The core of the changes looks fine. Just a couple of unrelated changes need to be removed from the PR.

**Comment by TheInvincibleRalph** [comment]
Done, @hanzei. Thank you!

File: package-lock.json
Comment (hanzei) [reply package-lock.json:1]: @TheInvincibleRalph The changes are still here :smiley:

Comment (TheInvincibleRalph) [reply package-lock.json:1]: ah.

File: server/channels/api4/saml.go [hunk 5]
```diff
@@ -107,6 +109,7 @@ func addSamlPrivateCertificate(c *Context, w http.ResponseWriter, r *http.Reques
 
 	auditRec := c.MakeAuditRecord("addSamlPrivateCertificate", audit.Fail)
 	defer c.LogAuditRec(auditRec)
+
```
Comment (hanzei) [inline server/channels/api4/saml.go:112]: This is also an unrelated changed

File: server/channels/api4/saml.go [hunk 6]
```diff
@@ -136,6 +139,7 @@ func addSamlIdpCertificate(c *Context, w http.ResponseWriter, r *http.Request) {
 
 	auditRec := c.MakeAuditRecord("addSamlIdpCertificate", audit.Fail)
 	defer c.LogAuditRec(auditRec)
+
```
Comment (hanzei) [inline server/channels/api4/saml.go:142]: Same here

File: package-lock.json
Comment (TheInvincibleRalph) [reply package-lock.json:1]: Oh. I have deleted it.

**Review by hanzei — CHANGES_REQUESTED** [review]
two more changes to go :rocket:

**Comment by TheInvincibleRalph** [comment]
All changes effected. Thank you for the review, @hanzei.

**Review by hanzei — APPROVED** [review]
LGTM :+1:

**Review by BenCookie95 — APPROVED** [review]
LGTM 🎉  Thank you!
